File: app/src/main/java/com/glucocontrol/domain/usecase/reading/DeleteReadingUseCase.kt
package com.glucocontrol.domain.usecase.reading

import com.glucocontrol.domain.repository.GlucoseRepository
import javax.inject.Inject

/**
 * Elimina permanentemente una lectura de glucosa identificada por su id.
 *
 * La operación es irreversible. La UI debe solicitar confirmación antes de invocar
 * este caso de uso (responsabilidad de la capa de presentación).
 */
class DeleteReadingUseCase
@Inject
constructor(private val repository: GlucoseRepository) {
    suspend operator fun invoke(id: Long) {
        require(id > 0) { "El id proporcionado no es válido: $id" }
        repository.deleteReading(id)
    }
}

File: lib/domain/usecase/reading/delete_reading_use_case.dart
import 'package:glucocontrol/domain/repository/glucose_repository.dart';

class DeleteReadingUseCase {
  final GlucoseRepository _repo;
  DeleteReadingUseCase(this._repo);

  Future<void> call(int id) async {
    if (id <= 0) throw ArgumentError('ID inválido');
    return _repo.deleteReading(id);
  }
}
